### PR TITLE
Snowpack

### PR DESCRIPTION
**Pull request by JoviDeCroock**
Depends on: https://github.com/pikapkg/snowpack/pull/332 

Just like `vite` at this moment it's lacking support for HMR on hooks due to the AST-part missing. I'll try and see if we can add this soon'ish.

To test this locally I added `react-refresh/babel` to my `babel.config.json`

### CODE REVIEW

**Comment by FredKSchott** [comment]
Looks good! This API just went out with `snowpack@2.0.0-beta.24` (`npm install snowpack@next`). Let me know if I can help at all with the integration.

File: packages/snowpack/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+npm i -s @prefresh/snowpack
+## OR
+yarn add @prefresh/snowpack
+```
+
+Then add it to your `snowpack.config.json` config:
+
+```json
+{
+  "plugins": ["@prefresh/snowpacks"]
```
Comment (FredKSchott) [inline packages/snowpack/README.md:19]: Typo :)

File: packages/snowpack/src/index.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                      compareSignaturesForPrefreshment(
+                        $CurrentModule$[i],
+                        module[i]
+                      );
+                    }
+                  }
+                }
+                $CurrentModule$ = module;
+              } catch(e) {
+                window.location.reload();
```
Comment (FredKSchott) [inline packages/snowpack/src/index.js:54]: Would it help to replace this with an import.meta.hot.reject() call instead? Feels lame to force you to reload yourself.

Comment (JoviDeCroock) [reply packages/snowpack/src/index.js:54]: Hmm, recently I discovered [this one](https://webpack.js.org/api/hot-module-replacement/#invalidate) which to me felt likea good course of action.

Comment (FredKSchott) [reply packages/snowpack/src/index.js:54]: Nice, +1 for matching their terminology.

**Comment by JoviDeCroock** [comment]
@FredKSchott I don't know if this is an issue on how I'm using it but I'm not seeing `babel` run atm initially I just added it to the `preact-scripts` found in the starter for `Preact`.

This didn't work so I made a local `babel.config.json` and added `"plugin:js,jsx": "@snowpack/plugin-babel"` to my package.json. My logs in the babel-plugin don't seem to trigger.

**Comment by FredKSchott** [comment]
Ah yea, we just finalized the interface:

- we use esbuild internally so no Babel needed, works with Preact & JSX, you can see the new preact CSA template to see that in action
- I can see how thats not obvious tho, we should show that that’s the new default in our dev output
- “plugin:js,jsx” is now just “build:js,jsx” ... and it will be connected automatically for you by just putting the plugin in your “plugins” config
- I think I updated the docs on snowpack.dev with these updates but it’s still new so still in progress

**Comment by JoviDeCroock** [comment]
Merging while we work on the bubbling HMR api for snowpack!

**Pull request merged**